Denik CZ.js

```diff
@@ -0,0 +1,131 @@
+function detectWeb(doc, url) {
+	return "newspaperArticle";
+}
+
+function doWeb(doc, url) {
+	scrape(doc, url);
+}
+
+function scrape(doc, url) {
+	var newArticle = new Zotero.Item('newspaperArticle');
+	newArticle.url = url;
+	newArticle.title = ZU.trimInternal(ZU.xpathText(doc, '//div[@class = "dv3-clanek-left-fix"]/h1')).replace(/^,/, "");
+	var date = ZU.xpathText(doc, '(//div[@class="dv3-clanek-datum"])');
+	var actualDateStr = '';
+	
+	if (ZU.trimInternal(date).indexOf('dnes') != -1) {
+		var actualDate = new Date(); 
+		actualDate.setHours(actualDate.getHours() + 2);
+		actualDateStr = actualDate.getUTCFullYear() + '-' +
+		('00' + (actualDate.getUTCMonth()+1)).slice(-2) + '-' +
+		('00' + actualDate.getUTCDate()).slice(-2);
+		date = actualDateStr;
+	} else if (ZU.trimInternal(date).indexOf('včera') != -1) {
+		var actualDate = new Date();
+		actualDate.setDate(actualDate.getDate() - 1);
+		actualDateStr = actualDate.getUTCFullYear() + '-' +
+		('00' + (actualDate.getUTCMonth()+1)).slice(-2) + '-' +
+		('00' + actualDate.getUTCDate()).slice(-2);
+		date = actualDateStr;
+	} else {
+		var parts = ZU.trimInternal(date).split('.');
+		actualDateStr = parts[2] + '-' + ZU.lpad(parts[1], 0, 2) + '-' + ZU.lpad(parts[0], 0, 2);
+		
+	}
+	//Zotero.debug(ZU.trimInternal(date).match('16'));
+	newArticle.date = actualDateStr;
+	var teaser = ZU.xpathText(doc, '//p[@class="dv3-clanek-perex"]');
+	if (teaser != null) {
+		newArticle.abstractNote = Zotero.Utilities.trimInternal(teaser).replace(/^,\s*/, "");
+	}
+
+	//some authors are in /a, some aren't we need to distinguish to get this right
+	if (ZU.xpathText(doc, '//p[@class = "clanek-autor"]/a') != null) {
+		var xpath = '//p[@class = "clanek-autor"]/a';
+		
+	} else {
+		var xpath = '//p[@class = "clanek-autor"]';
+	}
+	
+	if (ZU.xpathText(doc, xpath).indexOf('Redakce') != -1) {
+		newArticle.creators.push(Zotero.Utilities.cleanAuthor(ZU.xpathText(doc, xpath), "author"));
+	} else {
+		var authors = ZU.xpath(doc, xpath);	
+		for (var i=0; i<authors.length; i++) {
+			newArticle.creators.push(Zotero.Utilities.cleanAuthor(authors[i].textContent, "author"));
+		}
+	}
+	
+	
+	
+	var titleOfDenik = ZU.xpathText(doc, '//a[@class = "dv3-hlavicka-logo"]/span');	
+	if (titleOfDenik) {
+		titleOfDenik = titleOfDenik.charAt(0).toUpperCase() + titleOfDenik.slice(1).toLowerCase()
+		completedTitleOfDenik = titleOfDenik + " Deník"
+	} else {
+		completedTitleOfDenik = "Deník.cz"
+	}
+	Zotero.debug(completedTitleOfDenik);	
+	
+	newArticle.publicationTitle = completedTitleOfDenik;
+	
+	
+	newArticle.attachments.push({
+		title: completedTitleOfDenik + " Snapshot",
+		document: doc
+	});
+
+	newArticle.complete();
+}
+
+/** BEGIN TEST CASES **/
+var testCases = [
```

Comment:
You should add more test cases that cover some of the if blocks above (the "today" and "tomorrow" dates are not going to be testable unfortunately). E.g., do you have a page where there is no teaser, or is that just a precaution? You should cover the various cases with authors you have above.